### ISSUE TITLE
Functions incorrectly accepted as an implementation of any Protocol that has a matching __call__

### ISSUE THREAD

**Issue by Tomaz-Vieira** [microsoft/pyright#1888]
**Describe the bug**
A function that matches a `Protocol`'s `__call__` signature is incorrectly assumed to implement the entire `Protocol`

**To Reproduce**
Run the following through pyright:

```py
# pyright: strict
from typing_extensions import Protocol

class MyProtocol(Protocol):
    def __call__(self) -> int:
        ...

    def some_method(self) -> int:
        ...

def some_function() -> int:
    return 123

def protocol_user(p: MyProtocol):
    print(p.some_method())

protocol_user(some_function) #runtime error

```

**Expected behavior**
A compiler error when calling `protocol_user(some_function)`. Mypy gives the following error:

`error: Argument 1 to "protocol_user" has incompatible type "Callable[[], int]"; expected "MyProtocol"`

**VS Code extension or command-line**
pyright extension version 1.1.140
command line tool version 1.1.141

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.142, which I just published. It will also be included in the next release of pylance.
